### DISCUSSION TITLE
Enzyme Dev Guide or minimal knowledgebase requirement for contribution

### DISCUSSION THREAD

**Discussion by ipcamit** [EnzymeAD/Enzyme#1115] [General]
Hi, I anticipate having some free time in coming months. I wanted to go more deeper in Enzyme devel. I believe it will be a huge boon to my upcoming projects.

Is there a "developer guide" for internals of Enzyme? I understand developing compiler extensions is non-trivial to say the least, but I was wondering if someone can list some form of single semester/course worth of "Compiler minimum" curriculum needed to actually develop Enzyme further (rather than limited to being a user).

My background would include HPC C++/Fortran experience, with intermediate familiarity with compiler concepts.

Sorry if this off topic.

**Comment by tgymnich** [comment]
There is no official developer guide but feel free to join our weekly developer meetings: https://mit.zoom.us/j/96000853439. Meeting times are announced in our mailing list https://groups.google.com/d/forum/enzyme-dev. Right now our meeting is on Tuesday.

**Reply by ipcamit** [reply]
Thank you for the kind offer. What would be the minimum chapters in text book like "Engineering: A Compiler" (or any other recommended text) that I need to go through for participating in a meaningful way?
